### ISSUE TITLE
System.NotSupportedException - Unable to get member name of Parameter Expression

### ISSUE THREAD

**Issue by bslbckr** [agileobjects/AgileMapper#168]
We have a mapping scenario where we have to map from a inheritance hierarchy to a set of classes without inheritance allowed. 
this might look similar to this example:
```csharp
public enum SrcEnum {
	A,B,C,D,E,F,G,H,I
}

public enum TgtEnum {
	A,B,C,D,E,F,G,H,I
}

public interface ISomething {
	string Name {get;}
}
public class SomethingA : ISomething {
	public string Name {get;set;}
}
public class SomethingB : ISomething {
	public string Name {get;set;}
	public SrcEnum MyLargeEnum {get;set;}
}
public class TgtSomething {
	public string Name {get;set;}
	public TgtEnum MyLargeEnumg {get;set;}
}
```
The target-member `MyLargeEnum` can only be set from source, if the source-type is `SomethingB`  otherwise we don't have a matching counterpart. To solve this issue, we come up with the following mapping configuration:
```csharp
Func<IMappingData<ISomething, TgtSomething>,SrcEnum> getEnum =
			(ctxt) => {
			if (ctxt.Source is SomethingB) {
				var someB = (SomethingB) ctxt.Source;
				return someB.MyLargeEnum;
			}
			return SrcEnum.A;
		};
		Mapper.WhenMapping.From<ISomething>().To<TgtSomething>().Map(getEnum).To(tgt => tgt.MyLargeEnumg);
```
Unfortunately, this yields a run-time exception
```
[System.NotSupportedException: Unable to get member name of Parameter Expression]
   at AgileObjects.AgileMapper.Members.MemberExtensions.GetMemberName(Expression expression)
   at AgileObjects.AgileMapper.Validation.EnumMappingMismatchFinder.TryGetMatch(Expression targetMemberAccess, TargetMemberData& targetMemberData)
   at AgileObjects.AgileMapper.Validation.EnumMappingMismatchFinder.VisitBinary(BinaryExpression binary)
   at System.Linq.Expressions.BinaryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitBlock(BlockExpression node)
   at System.Linq.Expressions.BlockExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitBinary(BinaryExpression node)
   at AgileObjects.AgileMapper.Validation.EnumMappingMismatchFinder.VisitBinary(BinaryExpression binary)
   at System.Linq.Expressions.BinaryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitBlock(BlockExpression node)
   at System.Linq.Expressions.BlockExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitBlock(BlockExpression node)
   at System.Linq.Expressions.BlockExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitTry(TryExpression node)
   at System.Linq.Expressions.TryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitLambda[T](Expression`1 node)
   at System.Linq.Expressions.Expression`1.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at AgileObjects.AgileMapper.Validation.EnumMappingMismatchFinder.Process(Expression lambda, ObjectMapperData mapperData)
   at AgileObjects.AgileMapper.Plans.RootMapperMappingPlanFunction.GetDescription()
   at AgileObjects.AgileMapper.Plans.MappingPlan.<>c.<op_Implicit>b__3_0(IMappingPlanFunction pd)
   at AgileObjects.AgileMapper.Extensions.PublicEnumerableExtensions.<>c__0`2.<Project>b__0_0(Func`2 p, TItem item)
   at AgileObjects.AgileMapper.Extensions.PublicEnumerableExtensions.<Project>d__1`3.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at AgileObjects.AgileMapper.Plans.MappingPlan.op_Implicit(MappingPlan mappingPlan)
   at AgileObjects.AgileMapper.Plans.MappingPlan.ToString()
   at Program.Main() :line 41
```
The complete example could be found in this [fiddle](https://dotnetfiddle.net/tE0V3L)

**Comment by SteveWilkes** [comment]
This one turns out to be an error checking for enum value mismatches while translating the mapping plan into its readable format. The configuration you've used [does actually work](https://dotnetfiddle.net/pNfZ9S), provided you don't try to look at the mapping plan! (which of course I'll fix :) )

**Comment by bslbckr** [comment]
ok, that's good to know. 
We often use the mapping plan within our unit-tests in order to understand, why a mapping fails. With this bug our unit-test fails.

**Comment by SteveWilkes** [comment]
This is fixed in the latest v1.6 branch code - release to follow. Cheers! :)

**Comment by SteveWilkes** [comment]
This is fixed as of v1.6-preview4, which is now [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper). Thanks again!

**Comment by SteveWilkes** [comment]
The fix for this is included in v1.6, which is now available [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper/1.6.0).

Thanks again for the feedback!
